github/Repository.py

```diff
@@ -2325,15 +2325,20 @@ def get_forks(self):
     def create_fork(self, organization=github.GithubObject.NotSet):
         """
         :calls: `POST /repos/{owner}/{repo}/forks <https://docs.github.com/en/rest/reference/repos#forks>`_
-        :param organization: string or "none" or "*"
+        :param organization: the name of the organization to fork into,
+                             otherwise forks in the current user's account
+        :type organization: github.Organization.Organization or str or None
         :rtype: :class:`github.Repository.Repository`
         """
-        assert organization is github.GithubObject.NotSet or isinstance(
-            organization, str
-        ), organization
         post_parameters = {}
-        if organization is not github.GithubObject.NotSet:
+        if isinstance(organization, Organization.Organization):
+            post_parameters["organization"] = organization.login
+        elif isinstance(organization, str):
             post_parameters["organization"] = organization
+        else:
+            assert organization is None \
```

Comment:
Why? Everywhere else in this code base we use NotSet. I know None is more Pythonic, but we don't need extra (untested!) code to handle it

Reply:
Mostly because I figured it was not much and it would help caller code: currently if the caller can have an optional orgname (much like this method itself) it will have to do something like
```python
if name is None:
    repo.create_fork()
else:
    repo.create_fork(name)
```
or
```python
repo.create_fork(github.GithubObject.NotSet if name is None else name)
```
which seems even worse.